apps/extension/src/pages/formatting.tsx

```diff
@@ -0,0 +1,287 @@
+import { DateFormat, IntroSentenceType } from "@agreeto/api/types";
+import {
+  DEFAULT_LANGUAGE_FORMAT,
+  LANGUAGE_FORMATS,
+  type LanguageFormatItem,
+  extractTextFromSlots,
+} from "@agreeto/calendar-core";
+import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
+import { addHours, format, startOfDay } from "date-fns";
+import { useEffect, useRef, useState } from "react";
+import { HiCheckCircle } from "react-icons/hi";
+import { TiArrowSortedDown } from "react-icons/ti";
+
+import { trpcApi } from "~features/trpc/api/hooks";
+
+// @ts-expect-error - svg's are not typed
+import previewImage from "./../../assets/gmail-preview.svg";
+
+export const Formatting = () => {
+  const utils = trpcApi.useContext();
+  const introSentenceRef = useRef<HTMLTextAreaElement>(null);
+  const [languageDropdownOpen, setLanguageDropdownOpen] = useState(false);
+  const [dateFormatDropdownOpen, setDateFormatDropdownOpen] = useState(false);
```

Comment:
what are these used for? radix handles the open/close state itself?

Reply:
They are for highlighting parts in the preview. I copied and pasted from the old repo and apparently some CSS are broken. Normally those states update specific text color like in the screenshot taken from old app:

<img width="413" alt="Screen Shot 2022-12-01 at 10 19 30" src="https://user-images.githubusercontent.com/30385360/204990465-b5a7d083-81e4-4ad0-a4e2-5463f5ee7988.png">
<img width="423" alt="Screen Shot 2022-12-01 at 10 19 47" src="https://user-images.githubusercontent.com/30385360/204990519-848659ef-991a-48bd-9392-cb930954b011.png">